virtual_rainforest/abiotic/radiation.py

```diff
@@ -0,0 +1,179 @@
+"""The `abiotic.radiation` module.
+
+The radiation module calculates the radiation balance of the Virtual Rainforest.
+Top of canopy net shortwave radiation at a given location depends on
+    1. extra-terrestrial radiation (affected by the earth's orbit, time of year and
+        day, and location on the earth),
+    2. terrestrial radiation (affected by atmospheric composition and clouds),
+    3. topography (slope and aspect), and
+    4. surface albedo (vegetation type and fraction of vegetation/bare soil).
+
+Terrestial radiation can be derived from climatologies (e.g. WFDE5) or calculated here;
+the implementation is based on Davis et al., (2017). The effects of topography are taken
+into account with a factor derived from the preprocessing module.
+
+The vertical profile of net radiation below the canopy depends on the canopy
+structure, the amount of radiation absorbed by the canopy, and the longwave radiation
+emitted by canopy and surface.
+
+The radiation balance can be calculated with the radiation_balance() function. If
+terrestrial radiation is NOT provided as an input, it will be calculated internally.
+
+# the following structural components are not implemented yet
+TODO core.constants
+TODO include vertical structure (canopy layers)
+TODO include data structure (->numpy array)
+TODO include time dimension
+TODO logging, raise errors
+"""
+
+# import numpy as np
+# from core.constants import kc, kd, kfFEC, bolz # this doesn't exist yet
+kc = 1
+kd = 1
+kfFEC = 1
+bolz = 1
+
+
+class Radiation:
+    """Radiation balance.
+
+    Attributes:
+        topofcanopy_radiation: float, shortwave radiation, top of canopy [W m-2]
+        ppfd: float, photosynth. photon flux density, top of canopy [mol m-2]
+        rn_profile: float, net radiation profile, below canopy [W m-2]
+    """
+
+    def __init__(self) -> None:
+        """Initializes point-based radiation method."""
+        pass
+
+    def calc_sw_down(self, n: int, y: int, lat: float) -> float:
+        """Calculate shortwave downward radiation.
+
+        This function calculated incoming shortwave radiation based on location and time
+        of year if this is NOT provided as an input to the virtual rainforest.
+
+        Args:
+            n: day of the year
+            y: year
+            lat: latitude
+
+        Ref:
+            Davis et al. (2017): Simple process-led algorithms for simulating habitats
+            (SPLASH v.1.0): robust indices of radiation, evapotranspiration and plant-
+            available moisture, Geosci. Model Dev., 10, 689-708
+        """
+        raise NotImplementedError("Implementation of this feature is missing")
+
+    def calc_topofcanopy_radiation(
+        self,
+        sw_in: float,
+        n: int,
+        y: int,
+        lat: float,  # this is optional for the case of sw_in == FALSE
+        elev: float,
+        sf: float,
+        topo_factor: float,
+        sw_albedo: float,
+        vis_albedo: float,
```

Comment:
Some of these variables could have a more descriptive name (`shortwave_in`, `sunshine_hours`, `day`, `year`, etc.) . It does not save much space or time to shorten them and will make the code more explicit.